java/integTest/src/test/java/glide/modules/JsonTests.java

```diff
@@ -321,6 +321,95 @@ public void arrlen() {
         assertEquals(5L, res);
     }
 
+    @Test
+    @SneakyThrows
+    public void arrtrim() {
+        String key = UUID.randomUUID().toString();
+
+        String doc =
+                "{\"a\": [0, 1, 2, 3, 4, 5, 6, 7, 8], \"b\": {\"a\": [0, 9, 10, 11, 12, 13], \"c\": {\"a\":"
+                        + " 42}}}";
+        assertEquals("OK", Json.set(client, key, "$", doc).get());
+
+        // Basic trim
+        var res = Json.arrtrim(client, key, "$..a", 1, 7).get();
+        assertArrayEquals(new Object[] {7L, 5L, null}, (Object[]) res);
+
+        String getResult = Json.get(client, key, new String[] {"$..a"}).get();
+        String expectedGetResult = "[[1, 2, 3, 4, 5, 6, 7], [9, 10, 11, 12, 13], 42]";
+        assertEquals(JsonParser.parseString(expectedGetResult), JsonParser.parseString(getResult));
+
+        // Test end >= size (should be treated as size-1)
+        res = Json.arrtrim(client, key, "$.a", 0, 10).get();
+        assertArrayEquals(new Object[] {7L}, (Object[]) res);
+        res = Json.arrtrim(client, key, ".a", 0, 10).get();
+        assertEquals(7L, res);
+
+        // Test negative start (should be treated as 0)
```

Comment:
Redis and EC2 have different test results.
EC2 - result is 6
Redis - result is 0

Reply:
I think 6 is the right result, if negative start is treated as 0 (6 items in total from index 0 to index 5).